script/update-external-binaries.py

```diff
@@ -64,21 +73,37 @@ def is_updated(version_file, version):
   return existing_version == version
 
 
-def download_and_unzip(url_prefix, framework):
-  zip_path = download_framework(url_prefix, framework)
-  if zip_path:
-    extract_zip(zip_path, 'external_binaries')
+def binary_should_be_downloaded(binary):
+  if 'platform' in binary and binary['platform'] != PLATFORM:
+    return False
 
+  if 'arch' in binary and binary['arch'] != get_target_arch():
```

Comment:
i think we need a distinction between host and target arch here, since some of the binaries are to be linked with the target binary, and others are to be run on the host machine.

Reply:
Maybe rename the json property to `targetArch`?
```suggestion
  if 'targetArch' in binary and binary['targetArch'] != get_target_arch():
```

Reply:
we'll also need a `hostArch` property, because e.g. `sccache` is only run on the host.

Reply:
it's possible to build Electron ARM with an x86 machine though.

Reply:
ah i see, you're just saying that there's no binary currently that depends on the host architecture but not the host OS. yes, i agree.